### ISSUE TITLE
Support for quick edit buttons

### ISSUE THREAD

**Issue by danielbachhuber** [wp-shortcake/shortcake#169]
Core's embedded image representation has these nifty alignment quick-edit buttons:

![image](https://cloud.githubusercontent.com/assets/36432/6235133/0a95fec0-b695-11e4-89de-27bca251d4b9.png)

It would be neat if Shortcake:
- Optionally supported these alignment parameters for shortcodes
- Supporting registering arbitrary "Quick edit" shortcode buttons

**Comment by soderlind** [comment]
+1

**Comment by mattheu** [comment]
+:100: on the alignment

**Comment by soderlind** [comment]
@mattheu agree, and with alignment being able to set the width is need.

**Comment by mattheu** [comment]
The issue here however is the wp mceView doesn't really handle this at all...

**Comment by soderlind** [comment]
Hopefully the work [being done](https://make.wordpress.org/core/2015/01/22/the-case-for-js-modules/) will make it easier to grok wp.*

**Comment by dovy** [comment]
Clone anyone?  ;)

**Comment by danielbachhuber** [comment]
Bumping this out of v0.4.0. It needs a greater degree of technical clarity before we commit it to a milestone.

**Comment by cliffordp** [comment]
Could add custom ones too, like this: https://github.com/gambitph/Page-Builder-Sandwich/wiki/Adding-Toolbar-Buttons
![](https://github.com/gambitph/Page-Builder-Sandwich/wiki/images/toolbar.jpg)

**Comment by danielbachhuber** [comment]
For v0.5.0, if this isn't technically viable with a reasonable amount of development effort, we should at least establish what we need to do.

**Comment by bruno-barros** [comment]
wow guys. This PBS did an amazing job with shortcodes.
I dream with these two projects working together.

**Comment by cliffordp** [comment]
PBS relies on Shortcake

**Comment by bruno-barros** [comment]
Yes, of course, but Shortcake could bring some features from PBS like the contextual menu 
if it make sense.

**Comment by ivankristianto** [comment]
I want to bump this thread. Wondering if there is any progress on this feature?
